tests/unit/test_cv_device_tools.py

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/python
+# coding: utf-8 -*-
+# pylint: disable=logging-format-interpolation
+# pylint: disable=dangerous-default-value
+# pylint:disable=duplicate-code
+# flake8: noqa: W503
+# flake8: noqa: W1202
+# flake8: noqa: R0801
+
+from unittest.mock import call
+import pytest
+from tests.lib import mockMagic
+from tests.data.device_tools_unit import validate_ruter_bgp, validate_intf, validate_true
+from ansible_collections.arista.cvp.plugins.module_utils.device_tools import DeviceInventory, CvDeviceTools
+from ansible_collections.arista.cvp.plugins.module_utils.resources.modules.fields import ModuleOptionValues
+
+mock_m = mockMagic.MockModule()
+mockCvpClient = mockMagic.MockCvpClient()
+mockCvpApi = mockMagic.MockCvpApi()
+mockCvpClient.mock_cvpClient.api.validate_config_for_device.side_effect = mockCvpApi.validate_config_for_device
+
+device_data = [{
+    'serialNumber': '0123F2E4462997EB155B7C50EC148767',
+    'systemMacAddress': '50:08:00:b1:5b:0b',
+    'key': '50:08:00:b1:5b:0b',
+    'fqdn': 'tp-avd-leaf2',
+    'hostname': 'tp-avd-leaf2',
+    'parentContainerName': 'TP_LEAF1',
+    'configlets': ['']}] # this is dummy device_data that has no effect
+
+@pytest.mark.generic
+class TestValidateConfig():
+
+    # mocking
+    def apply_mocks(self, mocker):
+        mock_ansible_module = mock_m.apply_mock_patch(mocker,
+        'ansible_collections.arista.cvp.plugins.module_utils.device_tools.AnsibleModule')
+        mock__get_configlet_info = mock_m.apply_mock_patch(mocker,
+            'ansible_collections.arista.cvp.plugins.module_utils.device_tools.' \
+                'CvDeviceTools._CvDeviceTools__get_configlet_info')
```

Comment:
I feel this could be made simpler either by using 
```
        mock_ansible_module = mocker.patch(
        'ansible_collections.arista.cvp.plugins.module_utils.device_tools.AnsibleModule')
        mock__get_configlet_info = mocker.patch('ansible_collections.arista.cvp.plugins.module_utils.device_tools.'
                'CvDeviceTools._CvDeviceTools__get_configlet_info')
```

or maybe even with decorators)